src/core/lib.lua

```diff
@@ -374,8 +374,13 @@ if ffi.abi("be") then
    function htons(b) return b end
 else
   -- htonl is unsigned, matching the C version and expectations.
-   function htonl(b) return tonumber(cast('uint32_t', bswap(b))) end
-   function htons(b) return rshift(bswap(b), 16) end
+  -- Wrapping the return call in parenthesis avoids the compiler to do
+  -- a tail call optimization.  In LuaJIT when the number of successive
+  -- tail calls is higher than the loop unroll threshold, the
+  -- compilation of a trace is aborted.  If the trace was long that
+  -- can result in poor performance.
```

Comment:
You know, I think the problem was more that a trace can't end with a tail call to a builtin, or something like that.  We actually made tail calls *not* count towards the loop unroll threshold in #205.  But evidently they can still be a problem in some cases?  So weird.

Reply:
OK, I thought that was the reason. I'm OK with amending this message in a next PR.